### ISSUE TITLE
Lupdate can't pick some strings for translation due to missing context

### ISSUE THREAD

**Issue by d1vanov** [d1vanov/libquentier#63]
[QT_TR_NOOP](https://doc.qt.io/qt-5/qtglobal.html#QT_TR_NOOP) is heavily used across libquentier but it seems in some places it lacks the context required for translation. In particular, it happens when

1. `QT_TR_NOOP` is located within a macro; `lupdate` is not smart enough to figure out the context in this case, it's not a full fledged compiler with a preprocessor after all.
2. `QT_TR_NOOP` can only work with classes subclassing `QObject` and containing `Q_OBJECT` macro in their definition.

In such places either the class should be adjusted to become a full fledged `QObject` with `Q_OBJECT` macro or `QT_TRANSLATE_NOOP` macro should be used to provide the translation context explicitly.

**Comment by d1vanov** [comment]
Sample output from `lupdate` highlighting the problematic places:
```
cmake --build . --target lupdate
Scanning dependencies of target lupdate
Info: creating stash file C:\dev\libquentier\build\.qmake.stash
C:/dev/libquentier/src/enml/ENMLConverter_p.cpp:110: Class 'quentier::ENMLConverterPrivate' lacks Q_OBJECT macro
C:/dev/libquentier/src/enml/HTMLCleaner.cpp:34: Ignoring definition of undeclared qualified class
C:/dev/libquentier/src/local_storage/LocalStorageCacheManager_p.cpp:116: Class 'quentier' lacks Q_OBJECT macro
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:120: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:136: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:170: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:194: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:218: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:242: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:266: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:289: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:317: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:345: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:398: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:422: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:446: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:470: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:507: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:531: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:570: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:593: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:621: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:644: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:672: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:700: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:738: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:774: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:811: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:839: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:862: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:885: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:908: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:932: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:960: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:997: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1042: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1084: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1127: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1166: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1191: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1219: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1242: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1270: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1298: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1351: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1388: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1426: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1462: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1498: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1521: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1544: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1568: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1592: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1616: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1640: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1663: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1691: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1719: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1772: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1809: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1848: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1876: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManagerAsync.cpp:1900: Qualifying with unknown namespace/class ::LocalStorageManagerAsync
C:/dev/libquentier/src/local_storage/LocalStorageManager_p.cpp:348: Class 'quentier' lacks Q_OBJECT macro
C:/dev/libquentier/src/local_storage/LocalStorageManager_p.cpp:10356: Class 'quentier::T' lacks Q_OBJECT macro
C:/dev/libquentier/src/local_storage/LocalStorageManager_p.cpp:10850: Class 'quentier::TOrderBy' lacks Q_OBJECT macro
C:/dev/libquentier/src/local_storage/NoteSearchQueryData.cpp:304: Class 'quentier::NoteSearchQueryData' lacks Q_OBJECT macro
C:/dev/libquentier/src/local_storage/Transaction.cpp:46: Class 'quentier::Transaction' lacks Q_OBJECT macro
C:/dev/libquentier/src/note_editor/NoteEditor_p.cpp:147: tr() cannot be called without context
C:/dev/libquentier/src/note_editor/NoteEditor_p.cpp:157: tr() cannot be called without context
C:/dev/libquentier/src/note_editor/NoteEditor_p.cpp:7341: Class 'quentier' lacks Q_OBJECT macro
C:/dev/libquentier/src/note_editor/delegates/AddHyperlinkToSelectedTextDelegate.cpp:34: Class 'quentier' lacks Q_OBJECT macro
C:/dev/libquentier/src/synchronization/RemoteToLocalSynchronizationManager.cpp:80: tr() cannot be called without context
C:/dev/libquentier/src/synchronization/RemoteToLocalSynchronizationManager.cpp:85: tr() cannot be called without context
C:/dev/libquentier/src/synchronization/RemoteToLocalSynchronizationManager.cpp:91: tr() cannot be called without context
C:/dev/libquentier/src/synchronization/RemoteToLocalSynchronizationManager.cpp:96: tr() cannot be called without context
C:/dev/libquentier/src/synchronization/RemoteToLocalSynchronizationManager.cpp:1293: Class 'quentier::ElementType' lacks Q_OBJECT macro
C:/dev/libquentier/src/synchronization/RemoteToLocalSynchronizationManager.cpp:1864: Class 'quentier' lacks Q_OBJECT macro
C:/dev/libquentier/src/synchronization/RemoteToLocalSynchronizationManager.cpp:8633: Class 'quentier::PendingContainerType' lacks Q_OBJECT macro
C:/dev/libquentier/src/synchronization/UserStore.cpp:108: Class 'quentier::UserStore' lacks Q_OBJECT macro
C:/dev/libquentier/src/types/Note.cpp:160: Qualifying with unknown namespace/class ::Note
C:/dev/libquentier/src/types/Note.cpp:193: Qualifying with unknown namespace/class ::Note
C:/dev/libquentier/src/types/Notebook.cpp:130: Qualifying with unknown namespace/class ::Notebook
C:/dev/libquentier/src/types/Notebook.cpp:202: Qualifying with unknown namespace/class ::Notebook
C:/dev/libquentier/src/types/Resource.cpp:154: Qualifying with unknown namespace/class ::Resource
C:/dev/libquentier/src/types/SavedSearch.cpp:115: Qualifying with unknown namespace/class ::SavedSearch
C:/dev/libquentier/src/types/SavedSearch.cpp:189: Qualifying with unknown namespace/class ::SavedSearch
C:/dev/libquentier/src/types/Tag.cpp:114: Qualifying with unknown namespace/class ::Tag
C:/dev/libquentier/src/types/Tag.cpp:188: Qualifying with unknown namespace/class ::Tag
C:/dev/libquentier/src/types/User.cpp:134: Qualifying with unknown namespace/class ::User
C:/dev/libquentier/src/types/data/LinkedNotebookData.cpp:65: Class 'quentier::LinkedNotebookData' lacks Q_OBJECT macro
C:/dev/libquentier/src/types/data/NoteData.cpp:157: Class 'quentier::NoteData' lacks Q_OBJECT macro
C:/dev/libquentier/src/types/data/NotebookData.cpp:66: Class 'quentier::NotebookData' lacks Q_OBJECT macro
C:/dev/libquentier/src/types/data/SavedSearchData.cpp:62: Class 'quentier::SavedSearchData' lacks Q_OBJECT macro
C:/dev/libquentier/src/types/data/TagData.cpp:65: Class 'quentier::TagData' lacks Q_OBJECT macro
C:/dev/libquentier/src/utility/ApplicationSettings.cpp:32: Class 'quentier' lacks Q_OBJECT macro
C:/dev/libquentier/src/utility/EncryptionManager_p.cpp:67: Class 'quentier::EncryptionManagerPrivate' lacks Q_OBJECT macro
Updating '../translations/libquentier_ru_RU.ts'...
    Found 1303 source text(s) (0 new and 1303 already existing)
Built target lupdate
```

**Comment by d1vanov** [comment]
Fixed in commit [67518a2](https://github.com/d1vanov/libquentier/commit/67518a2307dfa8494c09d2d08dc5711eaa4f6fbc) in branch [development](https://github.com/d1vanov/libquentier/commit/development).
